From 35988cc4452f7df42e3c1b462731489bd33dade6 Mon Sep 17 00:00:00 2001
From: Chris Patterson <cpatterson@microsoft.com>
Date: Mon, 11 Sep 2023 16:56:06 -0400
Subject: [PATCH] sources: do not override datasource detection if None is in
 list

Users with datasource_list = [Azure, None] started failing to boot
properly outside of Azure with the changes to override datasource detection.

If the fallback "None" is included in the datasource_list, do not treat
the system as configured with a single datasource.

If users want to force a single datasource regardless of detection,
they can do so by removing None from the list.

Signed-off-by: Chris Patterson <cpatterson@microsoft.com>
---
 cloudinit/sources/__init__.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff -ruN a/cloudinit/sources/__init__.py b/cloudinit/sources/__init__.py
--- a/cloudinit/sources/__init__.py	2023-08-28 09:20:24.000000000 -0700
+++ b/cloudinit/sources/__init__.py	2023-09-13 15:00:23.287549869 -0700
@@ -352,10 +352,7 @@
                 self,
             )
             return True
-        elif self.sys_cfg.get("datasource_list", []) in (
-            [self.dsname],
-            [self.dsname, "None"],
-        ):
+        elif self.sys_cfg.get("datasource_list", []) == [self.dsname]:
             LOG.debug(
                 "Machine is configured to run on single datasource %s.", self
             )
